sage_modabvar/torsion_subgroup.py

```diff
@@ -289,18 +318,34 @@ def possible_orders(self):
             [1, 2, 4, 5, 10, 20]
         """
         try:
-            return self._possible_orders
+            if proof:
+                return self._possible_orders
+            else:
+                return self._possible_orders_proof_false
         except AttributeError:
             pass
 
-        # return the order of the cuspidal subgroup in the J0(p) case
         A = self.abelian_variety()
-        if A.is_J0() and A.level().is_prime():
-            self._order = QQ((A.level()-1)/12).numerator()
-            return [self._order]
+        N = A.level()
+        # return the order of the cuspidal subgroup in the J0(p) case
+        if A.is_J0() and N.is_prime():
+            self._possible_orders = [QQ((A.level()-1)/12).numerator()]
+            self._possible_orders_proof_false = self._possible_orders
+            return self._possible_orders
 
+        # the elliptic curve case
         if A.dimension() == 1:
-            return [A.elliptic_curve().torsion_order()]
+            self._possible_orders = [A.elliptic_curve().torsion_order()]
+            self._possible_orders_proof_false = self._possible_orders
```

Comment:
It should be fine because self._possible_orders_proof_false is only set when the order is explicitly known.